framework/doc/content/source/problems/SolutionInvalidity.md

```diff
@@ -0,0 +1,43 @@
+# SolutionInvalidity
+
+## Description
+
+The [/SolutionInvalidity.md] object holds solution invalid warning information for MOOSE. With this object, you can mark a solution as "invalid" and output which object and how many times the warning occurs. An invalid solution means that the solution somehow does not satisfy requirements, such as a value being out of the bounds of a correlation.  Solutions are allowed to be invalid *during* the nonlinear solve, but they are not allowed to be invalid once it converges. A "converged" solution that is marked as invalid will cause MOOSE to behave as if the solution did NOT converge - including cutting back timesteps, etc.
+
+To declare a solution as "invalid", use the macro in the following code to mark it. The user can provide a message to describe the invalidity:
+
+!listing /test/src/materials/NonsafeMaterial.C  re=\s+if \(_input_diffusivity > _threshold\).*?\}
+
+!alert tip
+It is recommended to have a unique message for each invalidity, especially when you want to mark multiple types of invalid solutions within one object.
+
+If any solution invalidity is detected during the solve, a summary table of solution invalid warnings will be generated and reported at the end of each time step. An example of this output is shown below:
+
+```
+Solution Invalid Warnings:
+------------------------------------------------------------------------------------------------------
+|     Object      | Latest | Timestep | Total |                        Message                       |
```

Comment:
I still dont like Latest. 
This is the latest residual computation right? so can we say "Converged" instead?

Reply:
The comment has been addressed